ports/samd/Makefile

```diff
@@ -175,8 +178,10 @@ endif
 all: $(BUILD)/firmware.uf2
 
 $(BUILD)/firmware.elf: $(OBJ)
+	$(ECHO) "PREPROCESS LINK $@"
+	$(Q)$(CC) -E -x c $(LDDEFINES) $(LD_FILES)| grep -v '^#' > $(BUILD)/link.ld
```

Comment:
Instead of using the C preprocessor, it's possible to define symbols using `--defsym`.  Eg stm32 does:
```
LDFLAGS += -Wl,--defsym=_estack_reserve=8
```
The nrf port also uses `defsym`.  I think it would be simpler to use that instead of CPP.

Reply:
I'll try that. Using the preprocessor was carried over from the MIMXRT port, but it's ugly. If using symdef works, it is more straight.